packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2740/onlp/builds/src/module/src/sysi.c

```diff
@@ -0,0 +1,201 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ *
+ * </bsn.cl>
+ ************************************************************
+ *
+ *
+ *
+ ***********************************************************/
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <onlplib/file.h>
+#include <onlp/platformi/fani.h>
+#include <onlp/platformi/ledi.h>
+#include <onlp/platformi/psui.h>
+#include <onlp/platformi/sysi.h>
+#include <onlp/platformi/thermali.h>
+#include "platform_lib.h"
+#include "x86_64_mlnx_msn2740_int.h"
+#include "x86_64_mlnx_msn2740_log.h"
+
+#define ONL_PLATFORM_NAME  "x86-64-mlnx-msn2740-r0"
+#define ONIE_PLATFORM_NAME "x86-64-mlnx_msn2740-r0"
+
+#define NUM_OF_THERMAL_ON_MAIN_BROAD  CHASSIS_THERMAL_COUNT
+#define NUM_OF_FAN_ON_MAIN_BROAD      CHASSIS_FAN_COUNT
+#define NUM_OF_PSU_ON_MAIN_BROAD      2
+#define NUM_OF_LED_ON_MAIN_BROAD      6
+
+#define COMMAND_OUTPUT_BUFFER        256
+
+#define PREFIX_PATH_ON_CPLD_DEV          "/bsp/cpld"
+#define NUM_OF_CPLD                      3
+static char arr_cplddev_name[NUM_OF_CPLD][30] =
+{
+    "cpld_brd_version",
+    "cpld_mgmt_version",
+    "cpld_port_version"
+};
+
+const char*
+onlp_sysi_platform_get(void)
+{
+    return ONL_PLATFORM_NAME;
+}
+
+int
+onlp_sysi_platform_info_get(onlp_platform_info_t* pi)
+{
+    int   i, v[NUM_OF_CPLD]={0};
+
+    for (i=0; i < NUM_OF_CPLD; i++) {
+        v[i] = 0;
+        if(onlp_file_read_int(v+i, "%s/%s", PREFIX_PATH_ON_CPLD_DEV, arr_cplddev_name[i]) < 0) {
+            return ONLP_STATUS_E_INTERNAL;
+        }
+    }
+    pi->cpld_versions = aim_fstrdup("brd=%d, mgmt=%d, port=%d", v[0], v[1], v[2]);
+
+    return ONLP_STATUS_OK;
+}
+
+void
+onlp_sysi_platform_info_free(onlp_platform_info_t* pi)
+{
+    aim_free(pi->cpld_versions);
+}
+
+
+int
+onlp_sysi_oids_get(onlp_oid_t* table, int max)
+{
+    int i;
+    onlp_oid_t* e = table;
+    memset(table, 0, max*sizeof(onlp_oid_t));
+
+    /* 6 Thermal sensors on the chassis */
+    for (i = 1; i <= NUM_OF_THERMAL_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_THERMAL_ID_CREATE(i);
+    }
+
+    /* 6 LEDs on the chassis */
+    for (i = 1; i <= NUM_OF_LED_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_LED_ID_CREATE(i);
+    }
+
+    /* 2 PSUs on the chassis */
+    for (i = 1; i <= NUM_OF_PSU_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_PSU_ID_CREATE(i);
+    }
+
+    /* 4 Fans and 2 PSU fans on the chassis */
+    for (i = 1; i <= NUM_OF_FAN_ON_MAIN_BROAD; i++)
+    {
+        *e++ = ONLP_FAN_ID_CREATE(i);
+    }
+
+    return 0;
+}
+
+int
+onlp_sysi_onie_info_get(onlp_onie_info_t* onie)
+{
+    int rv = onlp_onie_read_json(onie,
```

Comment:
How does this file get populated? Normally this API is the source of the information, not the consumer of it.